src/CommonLib/LdapProducerQueryGenerator.cs

```diff
@@ -0,0 +1,151 @@
+using System.Collections.Generic;
+using System.Linq;
+using SharpHoundCommonLib.Enums;
+using SharpHoundCommonLib.LDAPQueries;
+
+namespace SharpHoundCommonLib;
+
+public class LdapProducerQueryGenerator {
+    public static GeneratedLdapParameters GenerateDefaultPartitionParameters(CollectionMethod methods) {
+        var filter = new LdapFilter();
+        var properties = new List<string>();
+        
+        properties.AddRange(CommonProperties.BaseQueryProps);
+        properties.AddRange(CommonProperties.TypeResolutionProps);
+
+        if (methods.HasFlag(CollectionMethod.ObjectProps) || methods.HasFlag(CollectionMethod.ACL) || methods.HasFlag(CollectionMethod.Container)) {
```

Comment:
Almost feels like there's a pattern here to exercise but every time I think about trying to simplify this code I come up with something more complex than the boilerplate here. No change, just pontificating.